### PR TITLE
[feature](array_agg) support array_agg with param is array/map/struct

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
this pr we support array_agg function support param with array map struct type
Issue Number: close #xxx

### CODE REVIEW

File: be/src/vec/aggregate_functions/aggregate_function_collect.cpp [hunk 1]
```diff
@@ -69,15 +72,21 @@ AggregateFunctionPtr create_aggregate_function_collect_impl(const std::string& n
...
                 distinct, argument_types, result_is_nullable);
+    } else {
+        // generic serialize which will not use specializations, ShowNull::value always means array_agg
+        if constexpr (ShowNull::value) {
+            return do_create_agg_function_collect<void, HasLimit, ShowNull>(
+                    distinct, argument_types, result_is_nullable);
+        }
     }
 
     LOG(WARNING) << fmt::format("unsupported input type {} for aggregate function {}",
```
Comment (eldenmoon) [inline be/src/vec/aggregate_functions/aggregate_function_collect.cpp:92]: unreachable code? remove

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**